client/lib/plans/index.js

```diff
@@ -67,7 +67,33 @@ export function addCurrentPlanToCartAndRedirect( sitePlans, selectedSite ) {
 }
 
 export function getCurrentPlan( plans ) {
-	return find( plans, { currentPlan: true } );
+	const currentPlan = find( plans, { currentPlan: true } );
+
+	if ( currentPlan ) {
+		return currentPlan;
+	}
+
+	// the site doesn't have defined a current plan.
+	// Let's dive in its `plan` property
+	const site = sitesList.getSelectedSite();
+
+	if ( ! site ) {
+		return null;
+	}
+
+	const { plan } = site;
+
+	if ( ! plan ) {
+		return null;
+	}
+
+	// let's check if the current plan is the old `Enterprise` plan.
+	// if it is then let's assing `business` plan.
+	if ( plan.product_id === 1007 ) {
```

Comment:
we actually already have a `isEnterprise()` function (it's linked in my comment) and I'm fine with having that value encoded _somewhere_. I was just skeptical that _this_ is the place to do it. Moving it inside `isEnterprise()` would at least place it in a spot someone might start looking for if they needed.